src/main/java/com/getbase/http/HttpClient.java

```diff
@@ -52,38 +53,53 @@ public Response request(HttpMethod method,
                             Map<String, Object> params,
                             Map<String, String> headers,
                             String body) {
-        Request.Builder builder = new Request.Builder().
-                method(method).
-                url(this.config.getBaseUrl() + url);
+        setupMDC(method, url);
```

Comment:
According to logback docs http://logback.qos.ch/manual/mdc.html (_Advanced use_ section):

> (...) Given that MDC information is managed on a per thread basis, each thread will have its own copy of the MDC. Thus, there is no need for the developer to worry about thread-safety or synchronization when programming with the MDC because it handles these issues safely and transparently.